stdlib/public/core/DictionaryBridging.swift

```diff
@@ -620,13 +638,21 @@ extension _CocoaDictionary.Index {
 extension _CocoaDictionary.Index {
   @inlinable
   internal mutating func isUniquelyReferenced() -> Bool {
+    defer { _fixLifetime(self) }
+    guard _isNativePointer(_storage) else {
+      return false
+    }
+    unowned(unsafe) var storage = _bridgeObject(toNative: _storage)
     return _isUnique_native(&storage)
```

Comment:
This is quite broken; the result of `Index._asCocoa` is never unique, so checking uniqueness here will never succeed. This needs to be moved to the top-level index.